autorag/nodes/queryexpansion/run.py

```diff
@@ -0,0 +1,174 @@
+    filenames = list(map(lambda x: os.path.basename(x), filepaths))
+
+    summary_df = pd.DataFrame({
+        'filename': filenames,
+        'module_name': list(map(lambda module: module.__name__, modules)),
+        'module_params': module_params,
+        'execution_time': average_times,
+        **{metric: list(map(lambda result: result[metric].mean(), results)) for metric in strategies.get('metrics')},
+    })
+
+    # filter by strategies
+    if strategies.get('speed_threshold') is not None:
+        results, filenames = filter_by_threshold(results, average_times, strategies['speed_threshold'], filenames)
+    selected_result, selected_filename = select_best_average(results, strategies.get('metrics'), filenames)
+    best_result = pd.concat([previous_result, selected_result], axis=1)
+
+    # add summary.csv 'is_best' column
+    summary_df['is_best'] = summary_df['filename'] == selected_filename
+
+    # save the result files
+    best_result.to_parquet(os.path.join(save_dir, f'best_{os.path.splitext(selected_filename)[0]}.parquet'),
+                           index=False)
+    summary_df.to_parquet(os.path.join(save_dir, 'summary.parquet'), index=False)
+    return best_result
+
+
+def retrieval_by_retrieval_module(retrieval_module: Dict, resources_dir: str, decomposed_queries: List[List[str]],
+                                  top_k: int) -> Tuple[List[List[str]], List[List[float]]]:
+    module = Module.from_dict(retrieval_module)
+    retrieval_module_type = module.module_type
+
+    # set parameters for retrieval module
+    if retrieval_module_type == "bm25":
+        # check if bm25_path and file exists
+        bm25_path = os.path.join(resources_dir, 'bm25.pkl')
+        assert bm25_path is not None, "bm25_path must be specified for using bm25 retrieval."
+        assert os.path.exists(bm25_path), f"bm25_path {bm25_path} does not exist. Please ingest first."
+
+        # run retrieval function
+        bm25_corpus = load_bm25_corpus(bm25_path)
+        original_bm25 = bm25.__wrapped__
+        ids, scores = original_bm25(queries=decomposed_queries, bm25_corpus=bm25_corpus, top_k=top_k)
+
+    elif retrieval_module_type == "vectordb":
+        # check if chroma_path and file exists
+        chroma_path = os.path.join(resources_dir, 'chroma')
+        assert chroma_path is not None, "chroma_path must be specified for using vectordb retrieval."
+        assert os.path.exists(chroma_path), f"chroma_path {chroma_path} does not exist. Please ingest first."
+
+        # set embedding model
+        embedding_model_str = module.module_param["embedding_model"]
+        if embedding_model_str in embedding_models:
+            embedding_model = embedding_models[embedding_model_str]
+        else:
+            logger.error(f"embedding_model_str {embedding_model_str} does not exist.")
+            raise KeyError(f"embedding_model_str {embedding_model_str} does not exist.")
```

Comment:
Why do you set this things manually?
This kind of job already well done in retrieval node decorator. All you can just call retrieval module functions, and pass proper `project_dir`.